### PR TITLE
Implement slcli vlan edit functionality.

### PR DESCRIPTION
**Pull request by FernandoOjeda**
Implement slcli vlan edit functionalityhttps://github.com/softlayer/softlayer-python/issues/1282.

**slcli vlan edit --help**

Usage: slcli vlan edit [OPTIONS] IDENTIFIER

  Edit a vlan's details.

Options:
  -n, --name TEXT  The optional name for this VLAN
  -e, --note TEXT  The note for this vlan.
  -g, --tag TEXT   Tags to set e.g. "tag1,tag2", or empty string to remove all
  -h, --help       Show this message and exit.

The SoftLayer_Network_Vlan::editObject request allow to edit the note attribute but this is not shown by ui and api too.

### CODE REVIEW

**Comment by FernandoOjeda** [comment]
@allmightyspiff I got this error from analysis:

SoftLayer/managers/network.py:46:0: R0904: Too many public methods (37/35) (too-many-public-methods)

It seems that I can not increase more methods.

File: SoftLayer/managers/network.py [hunk 1]
```diff
@@ -685,3 +692,38 @@ def get_nas_credentials(self, identifier, **kwargs):
         """
         result = self.network_storage.getObject(id=identifier, **kwargs)
         return result
+
+    def edit(self, instance_id, name=None, note=None, tags=None):
+        """Edit a vlan.
+
+        :param integer instance_id: the instance ID to edit.
+        :param string name: valid name.
+        :param string note: note about this particular v;am.
```
Comment (allmightyspiff) [inline SoftLayer/managers/network.py:701]: should be: `:param string note: note about this particular vlan.`

File: SoftLayer/CLI/vlan/edit.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        'name': name,
+        'note': note
+    }
+
+    if tags:
+        data['tags'] = ','.join(tags)
+
+    mgr = SoftLayer.NetworkManager(env.client)
+    vlan_id = helpers.resolve_id(mgr.resolve_vlan_ids, identifier, 'VLAN')
+    vlan = mgr.edit(vlan_id, **data)
```
Comment (allmightyspiff) [inline SoftLayer/CLI/vlan/edit.py:35]: I'd prefer to be explicit here, as I think that is a little more readable.


```
    new_tags = None

    if tags:
        new_tags = ','.join(tags)

    mgr = SoftLayer.NetworkManager(env.client)
    vlan_id = helpers.resolve_id(mgr.resolve_vlan_ids, identifier, 'VLAN')
    vlan = mgr.edit(vlan_id, name=name, note=note, tags=new_tags)
```

File: SoftLayer/managers/network.py (hunk 1 above)
Comment (FernandoOjeda) [reply SoftLayer/managers/network.py]: Done.

File: SoftLayer/CLI/vlan/edit.py (hunk 2 above)
Comment (FernandoOjeda) [reply SoftLayer/CLI/vlan/edit.py]: Done.

**Pull request merged**